features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/telemetry/MetricBase.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.telemetry
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.sampling.RateBasedSampler
+import com.datadog.android.core.sampling.Sampler
+
+// Base class for metric events
+internal interface MetricBase {
+    fun sendMetric(isSuccessful: Boolean)
+
+    companion object {
+        // Basic Metric type key.
+        internal const val METRIC_TYPE = "metric_type"
+
+        fun startMetric(
```

Comment:
this method looks quite unusual here bundled in this interface as a companion object. Instead we could have these methods added in the `MetricsDispatcher` interface and then have a global `DatadogMetricsDispatcher(batchMetricsDispatcher, methodCalledMetricsDispatcher)` which delegates under the hood to the bundled components. What do you guys think ? @xgouchet @jonathanmos @0xnm